components/core/Tutorial/Header.tsx

```diff
@@ -0,0 +1,100 @@
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import tutorials from 'pages/tutorials/developing/_meta.en-US.json';
+import { Tutorial } from 'types/tutorial';
+
+export function getTutorialByRouteName(
+  json: Record<string, Tutorial>,
+  routeName: string,
+): Tutorial | null {
+  const hashTable: Record<string, Tutorial> = {};
+
+  // Create hash table based on route/href name - It is a faster than loop implementation
+  for (let key in json) {
+    const item = json[key];
+    if (item && item.href) {
+      hashTable[item.href] = item;
+    }
+  }
+
+  return hashTable[routeName] || null;
+}
+
+export default function Header() {
+  const router = useRouter();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [tutorial, setTutorial] = useState<Tutorial | null>(null);
+
+  const ogImage = `http://localhost:3000/api/og?title=${tutorial?.title}&category=${tutorial?.category}`;
+
+  const metadata = {
+    title: tutorial?.longTitle,
+    openGraph: {
+      title: tutorial?.longTitle,
+      type: 'article',
+      url: `https://docs.livepeer.org/${router.asPath}`,
+      images: [
+        {
+          url: ogImage,
+        },
+      ],
+    },
+    twitter: {
+      card: 'summary_large_image',
+      title: tutorial?.title,
+      images: [ogImage],
+    },
+  };
+
+  useEffect(() => {
+    const fetchedTutorial = getTutorialByRouteName(tutorials, router.asPath);
+    setTutorial(fetchedTutorial);
+    setIsLoading(false);
+  }, [router.asPath]);
+
+  if (isLoading || !tutorial) {
+    return;
+  }
+
+  return (
+    <section className="mt-5">
+      <script type="application/ld+json">{JSON.stringify(metadata)}</script>
```

Comment:
I don't think this is the structure of ld+json - there's a bunch of @ symbols and such. We have metadata already defined [here](https://github.com/livepeer/docs/blob/main/theme.config.tsx)

Reply:
Can you please elaborate what metadata should be included? I have taken Lee's blog as reference: https://github.com/leerob/leerob.io/blob/main/app/blog/%5Bslug%5D/page.tsx#LL34C10-L54C7

Reply:
Yeah, he's using the app directory which is a Next 13 feature. So we can't use that quite yet. It looks like he's using contentlayer and generating the structured data here:

https://github.com/leerob/leerob.io/blob/main/contentlayer.config.js#L22

And then referencing that in the blog post.

So we can definitely change this ld+json to reflect a similar structure to his, with `'@type': 'BlogPosting', ...`, and then also update the metadata here
https://github.com/livepeer/docs/blob/main/theme.config.tsx
to set the OpenGraph data